File: poo/Cpp/AprovadoExameReprovado/AprovadoExameReprovado.cpp
#include <iostream>

int main() {
    double m;
    int f;
    std::cin >> m >> f;
    if (m < 4.0 || f > 30) {
        std::cout << "Reprovado!" << std::endl;
    } else if (m >= 6.0) {
        std::cout << "Aprovado!" << std::endl;
    } else {
        std::cout << "Exame Final!" << std::endl;
    }
    return 0;
}

File: poo/Java/AprovadoExameReprovado/AprovadoExameReprovado.java
import java.util.Scanner;

public class AprovadoExameReprovado {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        double m = scanner.nextDouble();
        int f = scanner.nextInt();
        if (m < 4.0 || f > 30) {
            System.out.println("Reprovado!");
        } else if (m >= 6.0) {
            System.out.println("Aprovado!");
        } else {
            System.out.println("Exame Final!");
        }
        scanner.close();
    }
}